feat(config): managerFilePatterns (#34615)
renovatebot/renovate: docs/usage/modules/manager/index.md

### File Matching

> Most managers have a default `filePatterns` array.

Reviewer: ```suggestion
Most Renovate managers have a default `filePatterns` array.
```

> The `filePatterns` array has regular expression or glob pattern that matches against the repository file list.

Reviewer: ```suggestion
The `filePatterns` array has a regular expression, or glob pattern, that matches against the repository file list.
```

#### Managers with no default filePatterns

Some managers have no default `filePatterns`, because they have no filename convention that would let Renovate intelligently filter them.

> In such a case, the manager will be disabled until you create a `filePatterns` regular expression, e.g. like the following:

Reviewer: ```suggestion
If there is no default `filePattern`, the manager is disabled.
For the manager to work, you must create a `filePatterns` regular expression, or glob pattern.
For example:
```

Please double check that the `filePattern` accepts a glob pattern, before applying this suggestion. 😉

```json
{
  "kubernetes": {
    "filePatterns": ["/^config/.*\\.yaml$/"]
  }
}
```

#### Extending a manager's default filePatterns

> If the default `filePatterns` for a manager does not match against one of your relevant files, you can _extend_ the existing pattern(s) by configuring a manager's `filePatterns` like in this example:

Reviewer: ```suggestion
If the default `filePatterns` for a manager does not match your file(s), you can _extend_ the pattern.
You extend the pattern by configuring the manager's `filePatterns`.
For example:
```

```json
{
  "dockerfile": {

>     "filePatterns": ["does-not-look-like-a-docker-file"]

Reviewer: ```suggestion
    "filePatterns": ["a-weird-name-for-a-dockerfile"]
```

How about something like this? The example is about a user setting their own weird Dockerfile name.

Author: Both seems fine

  }
}
```

#### Ignoring files that match the default filePatterns

> Renovate will _extend_ the existing [`filePatterns`](../../configuration-options.md#filepatterns), meaning you don't need to include the default patterns like `Dockerfile` in your own array.

Reviewer: ```suggestion
Renovate _extends_ the existing [`filePatterns`](../../configuration-options.md#filepatterns).
This means do _not_ need to copy/paste the default patterns like `Dockerfile` in your own array.
```

Author: Former sounds better,

### File Matching

Most Renovate managers have a default `filePatterns` array.

> The `filePatterns` array has a regular expression, or glob pattern, that matches against the repository file list.

Reviewer 2: ```suggestion
The `filePatterns` array can contain regular expression or glob patterns, that match against the repository file list.
```

Reviewer: ```suggestion
The `filePatterns` array can hold a regular expression or glob pattern, that match against the repository file list.
```

Grammar fix for your suggestion above, plus change `contain` for `hold`.

#### Managers with no default filePatterns

Some managers have no default `filePatterns`, because they have no filename convention that would let Renovate intelligently filter them.
If there is no default `filePattern`, the manager is disabled.
For the manager to work, you must create a `filePatterns` regular expression, or glob pattern.
For example:

```json
{
  "kubernetes": {
    "filePatterns": ["/^config/.*\\.yaml$/"]
  }
}
```

#### Extending a manager's default filePatterns

If the default `filePatterns` for a manager does not match your file(s), you can _extend_ the pattern.
You extend the pattern by configuring the manager's `filePatterns`.
For example:

```json
{
  "dockerfile": {
    "filePatterns": ["does-not-look-like-a-docker-file"]
  }
}
```

#### Ignoring files that match the default filePatterns

Renovate will _extend_ the existing [`filePatterns`](../../configuration-options.md#filepatterns), meaning you don't need to include the default patterns like `Dockerfile` in your own array.
In other words, the patterns are "additive".

> If a manager matches a file that you _don't_ want it to, ignore it using the [`ignorePaths`](../../configuration-options.md#ignorepaths) configuration option.

Reviewer 2: ```suggestion
If a manager matches a file that you _don't_ want it to, ignore it using a negative match, e.g. `"filePatterns": ["!not-this-file.txt"]`.
```

### File Matching

Most Renovate managers have a default `filePatterns` array.
The `filePatterns` array has a regular expression, or glob pattern, that matches against the repository file list.

#### Managers with no default filePatterns

Some managers have no default `filePatterns`, because they have no filename convention that would let Renovate intelligently filter them.
If there is no default `filePattern`, the manager is disabled.
For the manager to work, you must create a `filePatterns` regular expression, or glob pattern.
For example:

```json
{
  "kubernetes": {

>     "filePatterns": ["/^config/.*\\.yaml$/"]

Reviewer 3: ```suggestion
    "filePatterns": ["config/**/*.yaml"]
```
looks simpler
